p8e-util/src/main/kotlin/io/p8e/crypto/SmartKeyAuthRefresh.kt

```diff
@@ -0,0 +1,34 @@
+package io.p8e.crypto
+
+import com.fortanix.sdkms.v1.api.AuthenticationApi
+import com.google.common.util.concurrent.ThreadFactoryBuilder
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit.MINUTES
+import kotlin.concurrent.thread
+
+class SmartKeyAuthRefresh(
```

Comment:
would it make more sense to just have this in p8e-shared and registered as an @Component, with an @Scheduled annotation on the refreshSmartKeyAuthToken to schedule it? (maybe with a guard to not do anything if there is no smartKey api key... which I guess we need to make nullable...) Then the SmartKeySigner wouldn't have to be responsible for kicking this off...

Reply:
ill make the change, seems like the better solution.